File: backend/src/main/java/com/team05/petmeeting/domain/user/dto/profile/UserFeedCommentRes.java
package com.team05.petmeeting.domain.user.dto.profile;

import com.team05.petmeeting.domain.comment.entity.FeedComment;
import com.team05.petmeeting.domain.feed.enums.FeedCategory;

import java.time.LocalDateTime;
import java.util.List;

public record UserFeedCommentRes(
        long totalCommentCount,
        List<ProfileCommentItem> comments
) {
    public static UserFeedCommentRes of(long totalCommentCount, List<FeedComment> commentList) {
        List<ProfileCommentItem> items = commentList.stream()
                .map(ProfileCommentItem::from)
                .toList();

        return new UserFeedCommentRes(totalCommentCount, items);
    }

    public record ProfileCommentItem(
            long feedId,
            FeedCategory category,
            String content,
            LocalDateTime createdAt,
            LocalDateTime updatedAt
    ) {

        public static ProfileCommentItem from(FeedComment comment) {
            return new ProfileCommentItem(
                    comment.getFeed().getId(),
                    comment.getFeed().getCategory(),
                    comment.getContent(),
                    comment.getCreatedAt(),
                    comment.getUpdatedAt()
            );
        }
    }
}

File: backend/src/main/java/com/team05/petmeeting/domain/user/dto/profile/UserFeedCommentRes.kt
package com.team05.petmeeting.domain.user.dto.profile

import com.team05.petmeeting.domain.comment.entity.FeedComment
import com.team05.petmeeting.domain.feed.enums.FeedCategory
import java.time.LocalDateTime


data class UserFeedCommentRes(
    val totalCommentCount: Long,
    val comments: MutableList<ProfileCommentItem>
) {

    data class ProfileCommentItem(
        val feedId: Long,
        val category: FeedCategory,
        val content: String,
        val createdAt: LocalDateTime,
        val updatedAt: LocalDateTime
    ) {
        companion object {
            @JvmStatic
            fun from(comment: FeedComment): ProfileCommentItem {
                return ProfileCommentItem(
                    comment.feed.id,
                    comment.feed.category,
                    comment.content,
                    comment.createdAt,
                    comment.updatedAt
                )
            }
        }
    }

    companion object {
        @JvmStatic
        fun of(totalCommentCount: Long, commentList: MutableList<FeedComment>): UserFeedCommentRes {
            val items = commentList.stream()
                .map { comment: FeedComment -> ProfileCommentItem.from(comment) }
                .toList()

            return UserFeedCommentRes(totalCommentCount, items)
        }
    }
}